### PR TITLE
alerter 4.0 (new formula)

### PR DESCRIPTION
**Pull request by hybras**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

alerter creates macOS notifications. I've never developed on macOS before though. I need help building / figuring out the build command. I have Xcode installed

### CODE REVIEW

File: Formula/alerter.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Alerter < Formula
+  desc "Send User Alert Notification on Mac OS X from the command-line"
+  homepage "https://github.com/vjeantet/alerter"
+  url "https://github.com/vjeantet/alerter/archive/refs/tags/004.tar.gz"
+  version "4"
+  sha256 "c4c16735e1a57ce04a5acfa762bd44ab8ef10884308725ab886b984b1de17bfe"
+  license "MIT"
+
+  def install
```
Comment (carlocab) [inline Formula/alerter.rb:9]:
```suggestion
  def install
    xcodebuild "-project", "alerter.xcodeproj", "-target", "alerter", "-configuration", "Release", "SYMROOT=build"
    bin.install "build/Release/alerter"
```
You probably need something like this. You should also have a `depends_on xcode: :build`.

Comment (hybras) [reply Formula/alerter.rb:11]: Thanks for this. I'm getting some signing error, need time to change the signing settings to manual or something

**Comment by hybras** [comment]
I'm not sure why CI is failing to build this. At risk of being smote, "it builds on my machine"

File: Formula/alerter.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Alerter < Formula
+  desc "Send User Alert Notification on Mac OS X from the command-line"
```
Comment (SMillerDev) [inline Formula/alerter.rb:2]:
```suggestion
  desc "Send User Alert Notification on macOS from the command-line"
```

File: Formula/alerter.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  license "MIT"
+
+  depends_on xcode: :build
+
+  def install
+    project_file = "alerter.xcodeproj/project.pbxproj"
+    old_identity = "CODE_SIGN_IDENTITY = \"Apple Development\";"
+    new_empty_identity = "CODE_SIGN_IDENTITY = \"-\";"
+    inreplace project_file, old_identity, new_empty_identity
+    xcodebuild "-project", "alerter.xcodeproj", "-target", "alerter", "-configuration", "Release", "SYMROOT=build"
```
Comment (carlocab) [inline Formula/alerter.rb:16]:
```suggestion
    xcodebuild, "-arch", Harware::CPU.arch,
               "-project", "alerter.xcodeproj",
               "-target", "alerter",
               "-configuration", "Release",
               "SYMROOT=build",
               "CODE_SIGN_IDENTITY="
```
The `-arch` flag should fix the CI failure (I think), and I think we can also do away with the `inreplace` in favour of of passing `CODE_SIGN_IDENTITY=` to `xcodebuild`. Try this out locally to see if it works first...

**Comment by carlocab** [comment]
There's a different problem on Mojave:
```
==> alerter -timeout 1 -title Alerter Test -message test
dyld: lazy symbol binding failed: Symbol not found: _objc_opt_new
  Referenced from: /usr/local/bin/alerter (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libobjc.A.dylib

dyld: Symbol not found: _objc_opt_new
  Referenced from: /usr/local/bin/alerter (which was built for Mac OS X 10.15)
  Expected in: /usr/lib/libobjc.A.dylib
```

File: Formula/alerter.rb (hunk 2 above)
Comment (hybras) [reply Formula/alerter.rb:2]: Nothing against this, i just copied alerter's own description

**Comment by hybras** [comment]
This formula is supposed to work on macOS 10.8+. I think this means some required lib was not statically linked, and was expected to be provided by the os

File: Formula/alerter.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  depends_on xcode: :build
+
+  def install
+    xcodebuild "-arch", Hardware::CPU.arch,
+                "-project", "alerter.xcodeproj",
+                "-target", "alerter",
+                "-configuration", "Release",
+                "SYMROOT=build",
+                "CODE_SIGN_IDENTITY=",
+                "MACOSX_DEPLOYMENT_TARGET=10.9"
```
Comment (carlocab) [inline Formula/alerter.rb:18]:
```suggestion
                "MACOSX_DEPLOYMENT_TARGET=#{MacOS.version}"
```

Comment (hybras) [reply Formula/alerter.rb]: Thank you... Is there a list of all the variables we can use somewhere

**Comment by hybras** [comment]
I think something is wrong with the latest [macOS 11 run ](https://github.com/Homebrew/homebrew-core/pull/81388/checks?check_run_id=3126696791). Its been queued / running for over 10 mins

**Comment by hybras** [comment]
Do I have anything left to do here?

**Review by carlocab — DISMISSED** [review]
You're all good. Nice work!

File: Formula/alerter.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class Alerter < Formula
+  desc "Send User Alert Notification on macOS from the command-line"
+  homepage "https://github.com/vjeantet/alerter"
+  url "https://github.com/vjeantet/alerter/archive/refs/tags/004.tar.gz"
+  version "4"
```
Comment (carlocab) [inline Formula/alerter.rb:5]: Actually, sorry, just noticed this. Do we need to explicitly specify the version like this? Isn't the version `004`?

Comment (hybras) [reply Formula/alerter.rb:5]: Look at [alerter's releases](https://github.com/vjeantet/alerter/releases), I assumed this was a quirky version of semantic versioning.

Comment (carlocab) [reply Formula/alerter.rb:5]: The tags all specify `00x`, so let's keep that for the version. We can just omit the `version` line I think.

Comment (hybras) [reply Formula/alerter.rb:5]: I'm confused. I though version was necessary. I used `brew create` and I thought it failed to parse the version, and that's why it wasn't included. Should I change or remove version?

Comment (hybras) [reply Formula/alerter.rb]: Oh it *inferred* the version from the url. My bad

**Review by carlocab — APPROVED** [review]
Thanks again for your work here!

**Pull request closed without merging**